### PR TITLE
Conversion dot op from LHLO to Affine

### PR DESCRIPTION
**Pull request by pashu123**
Add conversions from MLIR Lhlo dot op to affine loops. These conversions are run as part of -lhlo-legalize-to-affine pass.

### CODE REVIEW

**Comment by bondhugula** [comment]
Please address @joker-eph  comments from https://github.com/tensorflow/tensorflow/pull/40069 (the use of auto 42-45 isn't addressed).

File: tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc [hunk 1]
```diff
@@ -17,20 +17,74 @@ limitations under the License.
...
+                                PatternRewriter& rewriter) const override {
+    Value lhs = op.lhs();
+    Value rhs = op.rhs();
+    MemRefType lhs_type = lhs.getType().cast<MemRefType>();
+    MemRefType rhs_type = rhs.getType().cast<MemRefType>();
+    Type element_type = lhs_type.getElementType();
+    ArrayRef<int64_t> shape_lhs = lhs_type.getShape();
+    ArrayRef<int64_t> shape_rhs = rhs_type.getShape();
+
+    // Only matrices for LHS and RHS for now.
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc:48]: Is this comment out of place?

**Review by joker-eph — DISMISSED** [review]

File: tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc (hunk 1 above)
Comment (pashu123) [reply tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc:48]: Both rank-2 tensors and matrices signify the same thing. Looks redundant to me. Should i remove it?

Comment (gbaned) [reply tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc:48]: @joker-eph Can you please assist on above comments from @pashu123. Thanks!

Comment (joker-eph) [reply tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc:48]: Just remove it, the code is self-explanatory here.

Comment (pashu123) [reply tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc]: Done.

**Review by joker-eph — DISMISSED** [review]

File: tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h [hunk 2]
```diff
@@ -287,6 +287,30 @@ inline Value MapLhloOpToStdScalarOp<xla_lhlo::ConvertOp>(
...
+        loc, result_types, {lhs, rhs}, b);
+    return MapLhloOpToStdScalarOpImpl<FloatType, ::mlir::AddFOp>{}(
+        loc, result_types, {float_mul, result}, b);
+  }
+  if (element_type.isa<IntegerType>()) {
+    Value int_mul = MapLhloOpToStdScalarOpImpl<IntegerType, ::mlir::MulIOp>{}(
+        loc, result_types, {lhs, rhs}, b);
+    return MapLhloOpToStdScalarOpImpl<IntegerType, ::mlir::AddIOp>{}(
+        loc, result_types, {int_mul, result}, b);
+  }
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h:310]: Can you add a test for the integer branch?

Comment (pashu123) [reply tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h:310]: okay.

Comment (pashu123) [reply tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h:310]: Added the test case.

**Review by joker-eph — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Pull request merged**